### PR TITLE
Add PSA overview page

### PR DESCRIPTION
**Pull request by danny4478**

### CODE REVIEW

File: docs/api/security/psa.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+| NSPE         | Non-Secure Processing Environment   |
+| PSA          | Platform Security Architecture      |
+| RoT          | Root Of Trust                       |
+| SAU          | Security Attribution Unit           |
+| SPE          | Secure Processing Environment       |
+| SPM          | Secure Partition Manager            |
+
+### Overview
+
+Mbed PSA provides root of trust services and infrastructure for developing IoT applications.
```
[the remark this answers is not in the document]
Comment (danny4478) [reply docs/api/security/psa.md]: CC @ndevillard

File: docs/api/security/psa.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+- Asymmetric Multiprocessing (AMP) systems: Multicore ARMv7-M targets (for example, PSoC6 featuring CM4 and CM0+ cores). On these targets, one of the cores is dedicated to PSA use only and implements SPE. Mbed PSA provides PSA API proxy implementation on a nonsecure core, which redirects execution to the SPE.
+
+- ARMv8-M: Generation of ARM processors featuring TrustZone-M architecture. PSA support for this platforms is in final stages of development and will be added to the list of Mbed PSA supported platforms shortly.
+
+### Mbed PSA RoT services
+
+Mbed PSA provides the following services:
+
+- PSA RoT internal storage.
+- PSA Crypto APIs.
```
[the remark this answers is not in the document]
Comment (danny4478) [reply docs/api/security/psa.md]: CC @ndevillard

File: docs/api/security/psa.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+| RoT          | Root Of Trust                       |
+| SAU          | Security Attribution Unit           |
+| SPE          | Secure Processing Environment       |
+| SPM          | Secure Partition Manager            |
+
+### Overview
+
+Mbed PSA provides root of trust services and infrastructure for developing IoT applications.
+
+When Mbed OS is running on a PSA Security Model compliant target, Mbed PSA helps to protect cryptographic assets, credentials and critical code sections by providing an isolation between a Secure Processing Environment (SPE) and a Non-Secure Processing Environment (NSPE). The Secure Partition Manager (SPM), which uses the target's hardware features, manages the isolation. The SPM provides standardized IPC APIs that abstract the fact that partitions could be living inside a virtualized environment (v8M, TEE on Cortex-A) or inside another chip.
```
[the remark this answers is not in the document]
Comment (danny4478) [reply docs/api/security/psa.md]: It means that the same API is used regardless of the system architecture - the system can be a dual core system which separates the secure world and non-secure world by using different cores (for example, dual Cortex-v7M) or a single core which makes this separation by its TEE capabilities (for example, dual Cortex-v8M).
cc @alzix

File: docs/api/security/psa.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+## Mbed PSA
+
+### Terms and abbreviations
```
[the remark this answers is not in the document]
Comment (alzix) [reply docs/api/security/psa.md]: i'm not familiar with the process :) can you help on that?
@danny4478 can you add @AnotherButler as collaborator to your fork?

**Comment by AnotherButler** [comment]
Related: https://github.com/danny4478/Handbook/pull/3

File: docs/api/security/psa.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ndevillard) [reply docs/api/security/psa.md]: Calling something Mbed PSA is indeed confusing. Mbed implements some PSA APIs (Crypto, Storage), offers PSA FF APIs and a secure partition manager, but there is no real project called Mbed PSA which you can download and contribute to.
I suggest this is expanded to "The Mbed implementation of PSA", just to clarify.

**Comment by AnotherButler** [comment]
This PR is waiting on engineering approval.

**Comment by AnotherButler** [comment]
Depends on https://github.com/ARMmbed/mbed-os/pull/8745 and https://github.com/ARMmbed/mbed-os/pull/8744

**Comment by danny4478** [comment]
@AnotherButler The diagram link does not work anymore. Is this expected to work?
Please confirm that the png file is not needed anymore under security/png and I will remove the whole png directory.

**Comment by AnotherButler** [comment]
@danny4478 The link will work and the image will appear once the image is uploaded to our AWS. That's currently a manual process by the web team, so I'll put in a request to have them do that.

**Comment by danny4478** [comment]
@AnotherButler ARMmbed/mbed-os#8745 will not make it into 5.11. ARMmbed/mbed-os#8744 will probably be labeled as ready-to-merge by the end of the day.
Is this the last thing stopping this PR from being merged?

**Comment by danny4478** [comment]
@orenc17 Can you please review?

**Pull request merged**